[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51252 [nnc] Don't fuse intrinsics on cpu if they are slower
* #51246 [nnc] Don't use sleef where it's slower
* #51190 [nnc] Expose vectorized math functions to jit fuser.
* #51187 [nnc] Use sleef if its symbols are available
* **#51186 [nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs**
* #51125 [nnc] Refactor generation of intrinsics to reduce the amount of macro-hell

Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

bertmaher committed:
[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs

Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM

Krovatkin commented on torch/csrc/jit/tensorexpr/llvm_jit.cpp:
```diff
@@ -369,16 +343,19 @@ class TORCH_API PytorchLLVMJITImpl {
     std::unique_ptr<LLVMContext> C) {
-  impl_->addModule(std::move(M));
-  return Error::success();
+  return impl_->addModule(std::move(M), std::move(C));
```
nitpick: return value in a void func?

bertmaher committed:
Update on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

bertmaher committed:
Fix some compilation issues with gcc5.4, msvc; fix double dtype handling on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

bertmaher committed:
maybe fix windows on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_jit.h:
```diff
@@ -48,10 +48,12 @@ class TORCH_API PytorchLLVMJIT {
 
   JITSymbol findSymbol(const std::string Name);
 
+  bool hasSymbol(const std::string& Name);
```
This doesn't seem to be defined anywhere.

bertmaher committed:
Update on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

bertmaher committed:
Work around debug-mode llvm really not wanting you to lookup undefined symbols on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]

bertmaher committed:
fix sleef type suffix at the appropriate point in the stack on "[nnc][trivial] Refactor llvm_jit so the wrapper class doesn't depend on ifdefs"


Just a bit of drive-by cleanup; the wrapper class should be the same
for all builds so let's not conditionally compile it for no reason.

Differential Revision: [D26096205](https://our.internmc.facebook.com/intern/diff/D26096205/)

[ghstack-poisoned]
